### PR TITLE
Update ec2_pillar documentation to use metadata service

### PR DESCRIPTION
**Pull request by mr337**
### What does this PR do?
Clarify the documentation to include client side config for proper use of ec2_pillar module.

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/50818

### Tests written?

NO

### Commits signed with GPG?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/pillar/ec2_pillar.py [hunk 1]
```diff
@@ -14,6 +14,9 @@
 of the instances to specific hosts.  There is no glob matching at
 this time.
 
+  Note: If using `use_grain: True` minion must have `metadata_server_grains: True` set
+  in minion config file.
```
Comment (damon-atkins) [inline salt/pillar/ec2_pillar.py]: Please put a link to the meta data grain. Maybe with
 ```
:mod:`ec2 metadata <salt.grains.metadata>`
```
Maybe some from the salt team can provide a hint.

**Review by damon-atkins — CHANGES_REQUESTED** [review]

**Comment by mr337** [comment]
@damon-atkins I have made the recommended changes. I'll be the first to admit I'm not an expert on this so feel free to make more recommendations.

**Review by damon-atkins — APPROVED** [review]

**Comment by terminalmage** [comment]
This pull request should have been opened against an earlier branch, since develop would only get this documentation into the docs for the develop branch of Salt, and not for any of the current releases. Additionally, I found that the `metadata_server_grains` option lacked any documentation in the minion config file docs.

Therefore, I added the needed docs for that minion config option, and also took the suggested docs from this PR and added them to the ec2 pillar docs for the `2018.3` branch, where they will eventually be merged forward to develop.

Unfortunately that makes this PR obsolete, so I will have to close it. But thank you very much @mr337! You've helped make Salt's documentation better, and we are honored by your contribution.

**Comment by mr337** [comment]
@terminalmage Not a problem, thanks for correcting that info!

**Pull request closed without merging**